enterprise/reporting/reporting.go

```diff
@@ -336,6 +336,10 @@ func (*DefaultReporter) getAggregatedReports(reports []*types.ReportByStatus) []
 			EventType:      report.StatusDetail.EventType,
 			ErrorType:      report.StatusDetail.ErrorType,
 		})
+
+		if len(metricsByGroup[identifier].StatusDetails) >= r.maxReportsCountInARequest.Load() {
+			delete(metricsByGroup, identifier)
```

Comment:
instead of deleting, can add this check before adding to the map, delete is a costly operation on a map.

Reply:
By deleting entry from map we want to create a new batch request, how will adding this check before adding item will help?

Reply:
Okay got it.
@vamsikrishnakandi Instead of deleting the key, we can assign it a new value if the size exceeds the limit.

Reply:
idea here is, if there are more reports with same identifier that allowed per request, we want to split them into multiple requests. 

so we are deleting the value associated with the identifier when limit is reached. this way next time report comes with same identifier, metric is recreated and pushed to values.

Made the change as you suggested. Now, we create the metric, associate it with the identifier in the group, and push it to values either when the identifier is not found or when the maximum length for StatusDetails is reached (the old value associated with the identifier is replaced when the max length is reached).